bee-network/bee-autopeering/src/local/services.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::proto;
+
+use libp2p_core::multiaddr::Protocol;
+use serde::{Deserialize, Serialize};
+
+use std::{collections::HashMap, fmt, io, str::FromStr};
+
+/// Represents the name of a service.
+pub type ServiceName = String;
+pub(crate) type ServicePort = u16;
+
+/// The announced name of the autopeering service.
+pub const AUTOPEERING_SERVICE_NAME: &str = "peering";
+
+/// A mapping between a service name and its endpoint data.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub struct ServiceMap(HashMap<ServiceName, Service>);
+
+impl ServiceMap {
+    /// Registers a service with its bind address.
+    pub(crate) fn insert(&mut self, service_name: impl ToString, protocol: ServiceProtocol, port: ServicePort) {
+        self.0.insert(service_name.to_string(), Service { protocol, port });
+    }
+
+    /// Returns the connection data associated with the given service.
```

Comment:
Is this comment alright ?

Reply:
in German this couldn't be ambiguous, because we would just merge "connection" and "data" into "connectiondata" :trollface: 
The comment is alright :slightly_smiling_face:

Reply:
I found it a bit confusing because it returns a service, it's not "with the given service"

Reply:
I see....the changed the type name to "ServiceData" and changed the doc comment slightly.